implementations/federation-jvm/src/main/java/com/apollographql/federation/compatibility/graphqljava/GraphQLProvider.java

```diff
@@ -0,0 +1,88 @@
+package com.apollographql.federation.compatibility.graphqljava;
+
+import static graphql.schema.idl.TypeRuntimeWiring.newTypeWiring;
+
+import com.apollographql.federation.graphqljava.Federation;
+import com.apollographql.federation.graphqljava._Entity;
+import com.apollographql.federation.graphqljava.tracing.FederatedTracingInstrumentation;
+import graphql.GraphQL;
+import graphql.schema.GraphQLSchema;
+import graphql.schema.idl.RuntimeWiring;
+import graphql.schema.idl.SchemaParser;
+import graphql.schema.idl.TypeDefinitionRegistry;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.PostConstruct;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GraphQLProvider {
+
+    @Autowired
+    GraphQLDataFetchers graphQLDataFetchers;
+
+    GraphQLSchema graphQLSchema;
```

Comment:
is there any value in making `GraphQL`/`GraphQLSchema` properties on the class? if not I'd just initialize it within the `@Bean public GraphQL graphQL() {...}`, i.e. basically move the `init` logic there